jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/TreeSingleSourcePathsImpl.java

```diff
@@ -135,7 +135,7 @@ public double getWeight(V targetVertex)
         }
 
         double weight = 0d;
-        while (p != null && !p.equals(source)) {
+        while (p != null) {
```

Comment:
@jkinable The proposed fix in #489 is correct. We still need this check since there might be an entry also for the source vertex containing the tuple `(0, null)`. We should merge PR #489 instead.

Reply:
Ok, I stand corrected. Thanks for the clarification. Updated my branch accordingly.